module/VuFindTheme/src/VuFindTheme/ThemeInfo.php

```diff
@@ -210,35 +210,76 @@ public function getThemeInfo()
         return $this->allThemeInfo;
     }
 
+    /**
+     * Determine if a variable is a string-keyed array
+     *
+     * @param mixed $op Variable to test
+     *
+     * @return boolean
+     */
+    protected function isStringKeyedArray($op) {
+        if (!is_array($op)) {
+            return false;
+        }
+
+        reset($op);
+        return is_string(key($op));
+    }
+
+    /**
+     * Add parent theme information to a collection of merged theme info.
+     * Using the string-keyed array format of theme config info,
+     * recursively walk the array, capturing unique or missing values.
+     *
+     * @param array|string $merged Merged theme info, overrides parent theme
+     * @param array|string $current Theme info to be merged
+     *
+     * @return array|string merged theme info, favoring child themes (merged)
+     */
+    protected function mergeWithoutOverride($merged, $current) {
+        if (empty($merged)) {
+            return $current;
+        }
+
+        if ($this->isStringKeyedArray($merged)) {
+            foreach ($current as $key => $val) {
```

Comment:
I think we need to define a behavior for what happens when $merged is a string-keyed array and $current is a string... because otherwise we'll make an illegal foreach call and get a nasty error of some sort. Probably worth a test case!

Reply:
Thanks for the progress on this, @crhallberg, but is it really the case that when either $merged or $current is a non-array, we always want to return $merged? That does seem consistent with the method name of "mergeWithoutOverride," but don't we *want* to override string values in sub-themes? Or is the idea that we're applying these in reverse order from child to parent? If that's the case, maybe it is okay -- but I think this further reinforces my suggestion that we should be sure to have a test case that demonstrates this scenario, to be 100% clear about what we expect to happen.

Reply:
To return to my original comment, it looks to me like it may still be the case that things will go wrong if `$merged` is a string-keyed array and `$current` is not an array. Do we lack a test for that, or am I just missing/misunderstanding something?

Reply:
I believe these cases are covered here. https://github.com/vufind-org/vufind/pull/2543/files#diff-9dda0badc9ada5e653e0704a4e20ac71ebcb9fdc46c23c092442afa0d1909c43R364-R380

Reply:
@crhallberg, aren't those cases covering merging string-keyed and non-string-keyed arrays together? I'm talking about the case of merging a string-keyed array with a plain string (not an array at all). In other words, this test case:

```
            [
                [
                    ['mixed' => ['array']],
                    'not an array',
                ],
                ['mixed' => ['string', 'array']], // I'm not sure what we actually expect to happen in this case...
            ],
```

Reply:
@demiankatz I don't expect this case to ever show up since we're merging themeConfigs, but I've added a handle where it just skips the attempted override.